sdk/src/Core/Amazon.Util/Internal/PlatformServices/ServiceFactory.cs

```diff
@@ -92,5 +92,32 @@ private static Type GetServiceType<T>()
                 return _mappings[typeof(T)];
             }
         }
+
+        private static object CreateInstance(Type serviceType)
+        {
+            object service;
+            if (serviceType == typeof(IApplicationSettings))
+            {
+                service = new ApplicationSettings();
+            }
+            else if (serviceType == typeof(INetworkReachability))
+            {
+                service = new NetworkReachability();
+            }
+            else if (serviceType == typeof(IApplicationInfo))
+            {
+                service = new ApplicationInfo();
+            }
+            else if (serviceType == typeof(IEnvironmentInfo))
+            {
+                service = new EnvironmentInfo();
+            }
+            else
+            {
+                throw new ArgumentException($"Unknown service {typeof(T).FullName}");
```

Comment:
This doesn't build for me. 

`Error: The type or namespace name 'T' could not be found`

Should it be this instead?

```suggestion
                throw new ArgumentException($"Unknown service {serviceType.FullName}");
```

Reply:
Fixed